hbase-rsgroup/src/main/java/org/apache/hadoop/hbase/rsgroup/RSGroupBasedLoadBalancer.java

```diff
@@ -356,6 +357,11 @@ public void initialize() throws HBaseIOException {
     // Create the balancer
     Class<? extends LoadBalancer> balancerKlass = config.getClass(HBASE_RSGROUP_LOADBALANCER_CLASS,
         StochasticLoadBalancer.class, LoadBalancer.class);
+    if (balancerKlass.equals(this.getClass())) {
```

Comment:
`this.getClass().isAssignableFrom(balancerKlass)` this is what we can check here to stay consistent with master branch.

Reply:
Well, thank you

Reply:
Thanks @Gkkkk302
Once you are done with this change, will merge this for all branch-2s.